### PR TITLE
Fixed issue with unreported error on missing method call in if test.

### LINKED ISSUE #35557: No error for missing method call
**TypeScript Version:**  3.8.0-dev.20191206

**Search Terms:**
- diagnostics
- conditional
- missing function call

**Code**

```ts
import * as fs from 'fs';

fs.stat('/path/to/file', function(err, stats) {
    if (stats.isDirectory) { 
        console.log(`[Directory] ${stats.ctime}`)
    }
});
```

**Expected behavior:**
Error on `isDirectory`. This is a method but it has not been called

**Actual behavior:**
No error reported.

If I remove the console log line

### PR DESCRIPTION
**Pull request by dragomirtitian**
Fixed issue where missing method call went unreported if the call target symbol did no have an id assigned or if the called property was used inside the if block on a different target.

Fixes #35557

### CODE REVIEW

File: .vscode/settings.template.json [hunk 1]
```diff
@@ -1,19 +0,0 @@
-// Rename this file 'settings.json' or merge its
```
Comment (RyanCavanaugh) [inline .vscode/settings.template.json]: Please remove unrelated changes

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -30445,8 +30445,31 @@ namespace ts {
             const functionIsUsedInBody = forEachChild(ifStatement.thenStatement, function check(childNode): boolean | undefined {
                 if (isIdentifier(childNode)) {
                     const childSymbol = getSymbolAtLocation(childNode);
-                    if (childSymbol && childSymbol.id === testedFunctionSymbol.id) {
-                        return true;
+                    if (childSymbol && childSymbol === testedFunctionSymbol) {
+                        // If the test was a simple identifier, the above check is sufficient
+                        if(isIdentifier(ifStatement.expression)) {
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: Formatting in multiple places here is missing space between control flow keywords and `(`

File: .vscode/settings.template.json (hunk 1 above)
Comment (dragomirtitian) [reply .vscode/settings.template.json]: Sorry about this one, my bad.

File: src/compiler/checker.ts (hunk 2 above)
Comment (dragomirtitian) [reply src/compiler/checker.ts]: Shouldn't the linter have caught this ? The lint is clean though 😕

**Comment by sandersn** [comment]
This is a pretty old commit, so merging will probably break the build. On the other hand, I can't fetch commits from dragomirtitian/TypeScript. So I'm going to merge and immediately build locally and push any fixes to master.
